### ISSUE TITLE
AGS 4: Script API: add "split options" for String.Split method

### ISSUE THREAD

**Issue by ivan-mogilko** [adventuregamestudio/ags#2458]
I propose to add optional "split options" as a set of flags for the String.Split methods, equivalent or similar to the C# library function:
https://learn.microsoft.com/en-us/dotnet/api/system.stringsplitoptions?view=net-8.0

Something like
```
enum StringSplitOptions {
    eStrSplit_None = 0,
    eStrSplit_RemoveEmpty = 0x0001,
    eStrSplit_Trim = 0x0002
};

import String[] Split(const string separator, StringSplitOptions options = eStrSplit_None);
```

**Comment by AlanDrake** [comment]
I agree, it's a good convenience for several cases.

**Comment by ericoporto** [comment]
It misses the actual flags, but I believe the gist of the changes would be something like this

https://github.com/ericoporto/ags/commit/de5d29cd1d2f339dcaa8ca75336fe98b933a2e12

Not sure if there is a pretty way to get the trim in a single function.

**Comment by ericoporto** [comment]
Uhm, I wonder if a string like ` , ,  ,    ,` is passed with both flags, what the result should be, would it be null or does an empty array exists in AGS script?

**Comment by ivan-mogilko** [comment]
The trim implementation from above commit looks fine, similar to what I've imagined.

> Uhm, I wonder if a string like `" , , , ,"` is passed with both flags, what the result should be, would it be null or does an empty array exists in AGS script?

AGS does not currently allow a zero-length array. But if this is a matter of convenience, we may investigate if it's possible to support this.

**Comment by ericoporto** [comment]
I put a PR in #2461, please test it!

**Comment by ivan-mogilko** [comment]
Resolved by #2461
